app/models/Database.php

```diff
@@ -88,7 +88,37 @@ private function updateInnerTasks(string $id, string $parameter, string $input):
         $this->tasks = $updated_tasks;
     }
 
+    public function removeTask(string $id): array {
+        foreach($this->tasks as $key => $task) {
+            if ($task['id'] === $id) {
+                unset($this->tasks[$key]);
+                $this->updateDatabase();
+                break;
+            }
+        }
+        return $this->tasks;
+    }
+
+    public function addTask(array $newTask) {
+        $newTask['id'] = uniqid('');
+        do {
+            $taskExists = false;
+            foreach ($this->tasks as $task) {
+                if ($task['id'] === $newTask['id']) {
+                    $taskExists = true;
+                    $newTask['id'] = uniqid('');
+                    break;
+                }
+            }
+        } while ($taskExists);
+    
+        array_push($this->tasks, $newTask);
+        $this->updateDatabase();
+        return $this->tasks;
+    }
+
     private function updateDatabase(){
+        $this->tasks = array_values($this->tasks);
```

Comment:
Why this change?

Reply:
Because without this line the JSON was out of order:

Without code
![ex  without code](https://github.com/user-attachments/assets/22a9251a-b894-46bb-ad61-4040d1e816df)

With code
![ex  with code](https://github.com/user-attachments/assets/1f6c1e0b-7f2a-4cd2-ad85-ba75f185988f)

Reply:
But there's no need to order it, no? I thought our db was: 
<img width="606" alt="Screenshot 2024-11-21 at 17 05 58" src="https://github.com/user-attachments/assets/99b652fc-3642-4130-8cf5-5729a195795c">

Reply:
I mean, when I test it with editTask it gets correctly updated with and without that line... 🧐 

Maybe we can keep updateDatabase() method simple (it only handles copying "tasks" object into JSON file) and then formatting $this->tasks can be done in each method? 

In what method do you find these changes? Maybe I can test it out too

Reply:
It only happens when deleting a task. From what I saw, in PHP if you convert an array to JSON, if it doesn't have consecutive numeric indexes, it becomes a JSON object and changes the format, so it adds the indexes "1":{}. That's why I reindex the array with array_values, so that when rewriting the JSON the format isn't changed. It doesn't affect much in principle but it would constantly be changing the JSON format that we had agreed upon at the beginning.

Since it only happens with DELETE, I can pass it to its corresponding method as well.

Reply:
Oh, I see. Nono, then it's ok that it is here. Thanks for explaining :)